### PR TITLE
Add dataproc component gateway

### PR DESCRIPTION
**Pull request by c2thorn**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
[beta-only]dataproc: added component gateway support to `google_dataproc_cluster`
```
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/4128

~~Note:
The Component Gateway feature is in [alpha](https://cloud.google.com/dataproc/docs/concepts/accessing/dataproc-gateways), so I reached out to the Dataproc team to see if it would be a good idea to support it now. This PR will be available to anyone who wants it, but it won't be merged until we have a stronger sense of if the API will remain as it is now.~~

### CODE REVIEW

File: third_party/terraform/resources/resource_dataproc_cluster.go.erb [hunk 1]
```diff
@@ -353,6 +353,20 @@ func resourceDataprocCluster() *schema.Resource {
...
+						"endpoint_config": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enable_http_port_access": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
```
Comment (danawillow) [inline third_party/terraform/resources/resource_dataproc_cluster.go.erb]: This is totally fine to include, but also not necessary- it'll be false-by-default, by default

File: third_party/terraform/resources/resource_dataproc_cluster.go.erb [hunk 2]
```diff
@@ -674,6 +692,14 @@ func expandAutoscalingConfig(cfg map[string]interface{}) *dataproc.AutoscalingCo
 	}
 	return conf
 }
+
+func expandEndpointConfig(cfg map[string]interface{}) *dataproc.EndpointConfig {
+	conf := &dataproc.EndpointConfig{}
```
Comment (danawillow) [inline third_party/terraform/resources/resource_dataproc_cluster.go.erb]: I think you'll also want set `ForceSendFields: []string{"EnableHttpPortAccess"}` in order to guarantee that false values get sent

File: third_party/terraform/tests/resource_dataproc_cluster_test.go.erb [hunk 3]
```diff
@@ -582,6 +582,28 @@ func TestAccDataprocCluster_withNetworkRefs(t *testing.T) {
...
+func TestAccDataprocCluster_withEndpointConfig(t *testing.T) {
+	t.Parallel()
+
+	var cluster dataproc.Cluster
+	rnd := acctest.RandString(10)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		Steps: []resource.TestStep{
+			{
```
Comment (danawillow) [inline third_party/terraform/tests/resource_dataproc_cluster_test.go.erb]: If you add another step to modify the value from true to false, that should verify whether my comment above actually applies

File: third_party/terraform/resources/resource_dataproc_cluster.go.erb (hunk 2 above)
Comment (c2thorn) [reply third_party/terraform/resources/resource_dataproc_cluster.go.erb]: Turns out the field is only available on create, meaning no need for this

File: third_party/terraform/tests/resource_dataproc_cluster_test.go.erb (hunk 3 above)
Comment (c2thorn) [reply third_party/terraform/tests/resource_dataproc_cluster_test.go.erb]: Turns out the field is only available on create, meaning no need for this

**Comment by danawillow** [comment]
LGTM. I'm going to hold off on approving this until the feature is in beta so we don't accidentally merge this.

**Comment by c2thorn** [comment]
Updated these changes to fit with the current provider (3.0.0 changes, vcr, etc). Also added a computed only field.

**Pull request merged**